tests/integration/clientv3/grpc_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/v3rpc"
+	"go.etcd.io/etcd/tests/v3/integration"
+)
+
+func TestAuthoritySingleEndpoint(t *testing.T) {
+	integration.BeforeTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	kv := clus.Client(0)
+
+	ctx := context.TODO()
+	_, err := kv.Put(ctx, "foo", "bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	reqs := clus.Members[0].RecordedRequests()
+	assert.ElementsMatch(t, []v3rpc.RequestInfo{{FullMethod: "/etcdserverpb.KV/Put", Authority: "#initially=[unix://localhost:m00]"}}, reqs)
```

Comment:
How this behavies in case of communication over network instead of unix sockets, e.g. 
`localhost:8090`:

I would assume localhost:8090 should be an 'authority' in such case... 

In general IMHO we should default to the grpc behavior we would get
if we didn't override target in that line:

https://github.com/etcd-io/etcd/blob/97756e325cd9bee96fc28db7b2dc16974bf88539/client/v3/client.go#L302

i.e. keep it from this test perspective equivalent to the behavior for:

```
target := initialEndpoints[0]
```

Reply:
> How this behavies in case of communication over network instead of unix sockets, e.g.
> `localhost:8090`:

This is covered by e2e tests